File: docs/syntax-tests/syntax-test-2.c
#include <stdbool.h>
#include <stdio.h>

typedef struct Widget {
    const char *name;
    int count;
    bool enabled;
} Widget;

static int score_widget(const Widget *widget) {
    return widget->enabled ? widget->count * 2 : 0;
}

int main(void) {
    Widget widget = {.name = "syntax", .count = 7, .enabled = true};
    printf("%s => %d\n", widget.name, score_widget(&widget));
    return 0;
}

File: docs/syntax-tests/syntax-test-2.java
import java.util.List;

record Widget(String name, int count) {}

public final class SyntaxTestTwo {
    public static void main(String[] args) {
        List<Widget> widgets = List.of(new Widget("syntax", 2), new Widget("diff", 5));

        widgets.stream()
            .map(widget -> widget.name() + ":" + widget.count())
            .forEach(System.out::println);
    }
}
